packages/gatsby/src/redux/persist.ts

```diff
@@ -74,7 +89,7 @@ function guessSafeChunkSize(values: Array<[string, IGatsbyNode]>): number {
   }
 
   // Sends a warning once if any of the chunkSizes exceeds approx 500kb limit
-  if (maxSize > 500000) {
+  if (showMaxSizeWarning && maxSize > 500000) {
     report.warn(
       `The size of at least one page context chunk exceeded 500kb, which could lead to degraded performance. Consider putting less data in the page context.`
```

Comment:
We've talked about adding information about _what_ page exceeds this limit, but that's for a follow-up (not in this PR)